### PR TITLE
Always use the same renderer

### PR DESCRIPTION
**Pull request by mwestphal**
- The window now handle a single renderer (with coloring) to avoid recreating renderer which can have a very high cost when using HDRIs
-  Add changelog

### CODE REVIEW

File: library/src/window_impl.cxx [hunk 1]
```diff
@@ -50,10 +50,11 @@ class window_impl::internals
 
   std::unique_ptr<camera_impl> Camera;
   vtkSmartPointer<vtkRenderWindow> RenWin;
-  vtkSmartPointer<vtkF3DRenderer> Renderer;
+  vtkNew<vtkF3DRendererWithColoring> Renderer;
```
Comment (Meakk) [inline library/src/window_impl.cxx:53]: So `vtkF3DRenderer` is never instantiated now? Then maybe we should merge it with `vtkF3DRendererWithColoring`

Comment (mwestphal) [reply library/src/window_impl.cxx:53]: > So vtkF3DRenderer is never instantiated now?

Correct

> Then maybe we should merge it with vtkF3DRendererWithColoring

I don't think that is worth it. I like the conceptual separation.

Comment (Meakk) [reply library/src/window_impl.cxx:53]: fair enough

**Review by Meakk — APPROVED** [review]

**Comment by mwestphal** [comment]
CI is not clean though ^^

**Review by Meakk — APPROVED** [review]

File: library/src/window_impl.cxx [hunk 2]
```diff
@@ -205,36 +206,19 @@ point3_t window_impl::getDisplayFromWorld(const point3_t& worldPoint) const
...
-  {
-    this->Internals->Renderer = vtkSmartPointer<vtkF3DRendererWithColoring>::New();
-  }
-  else if (!withColoring && (renWithColor || !this->Internals->Renderer))
-  {
-    this->Internals->Renderer = vtkSmartPointer<vtkF3DRenderer>::New();
-  }
+  // Hide axis to make sure the renderer can be deleted if needed
+  this->Internals->Renderer->ShowAxis(false);
+  this->Internals->RenWin->RemoveRenderer(this->Internals->Renderer);
```
Comment (mwestphal) [inline library/src/window_impl.cxx:221]: this is actually wrong

**Pull request merged**